src/base/snark_intf.ml

```diff
@@ -1122,6 +1122,9 @@ module type Run_basic = sig
   (** The finite field over which the R1CS operates. *)
   type field
 
+  (* a reference to the internal Run_state.t *)
+  val state : field Run_state.t ref
```

Comment:
Would really prefer to keep this internal. And expose a pair of `get_state: unit -> field Run_state.t`, `set_state: field Run_state.t -> unit` functions. This permits any sort of checking to be implemented in there.

Reply:
oh you mean enables us to add checks on any state before it is used for example. that makes sense

Reply:
In no particular order:
1. easier to grep for `set_state` than `:=` 😄 
2. you can check for the value you set, or transform it as you wish, if you need to respect some internal invariants.